synapse/storage/databases/main/event_federation.py

```diff
@@ -1664,19 +1664,12 @@ async def get_successor_events(self, event_id: str) -> List[str]:
     @wrap_as_background_process("delete_old_forward_extrem_cache")
     async def _delete_old_forward_extrem_cache(self) -> None:
         def _delete_old_forward_extrem_cache_txn(txn: LoggingTransaction) -> None:
-            # Delete entries older than a month, while making sure we don't delete
-            # the only entries for a room.
```

Comment:
I've been trying to figure this out, but couldn't see any reason for it until last night.

I think it *may* have been to protect against the case of a room that hasn't been updated in the last month, where you want to return the most recent extremities, so that the caller then becomes a no-op.

Let me think how best to keep that optimisation.